File: grifballwebapp.client/src/app/api/dtos/signupRequestDto.ts
import { TimeslotDto } from "./signupResponseDto";

export class SignupRequestDto {
  seasonID: number = 0;
  userID: number = 0;
  teamName: string | null = null;
  willCaptain: boolean = false;
  requiresAssistanceDrafting: boolean = false;
  timeslots: TimeslotDto[] = [];
}

File: GrifballWebApp.Server/Signups/SignupRequestDto.cs
namespace GrifballWebApp.Server.Signups;

public class SignupRequestDto
{
    public int SeasonID { get; set; }
    public int UserID { get; set; }
    public string? TeamName { get; set; }
    public bool WillCaptain { get; set; }
    public bool RequiresAssistanceDrafting { get; set; }
    public TimeslotDto[] Timeslots { get; set; } = [];
}
